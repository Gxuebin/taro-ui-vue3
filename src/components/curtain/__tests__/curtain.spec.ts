import { mount } from '@vue/test-utils'
import { VNode } from '@vue/runtime-core'
import AtCurtain from '../index'

const factory = (
  values = {},
  slots: { default: Array<VNode | string> } = { default: [] }
) => {
  return mount(AtCurtain as any, {
    global: {
      components: {}
    },
    slots,
    props: { ...values },
  })
}

describe('AtCurtain', () => {
  it('should render default AtCurtain', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- class', () => {
    const wrapper = factory({ class: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- isOpened', () => {
    const wrapper0 = factory({ isOpened: true })
    expect(wrapper0.element).toMatchSnapshot()
    const wrapper1 = factory({ isOpened: false })
    expect(wrapper1.element).toMatchSnapshot()
  })

  it('should render prop -- closeBtnPosition', () => {
    const wrapper0 = factory({ closeBtnPosition: 'top' }, { default: ['test'] })
    expect(wrapper0.element).toMatchSnapshot()
    const wrapper1 = factory({ closeBtnPosition: 'bottom' }, { default: ['test'] })
    expect(wrapper1.element).toMatchSnapshot()
  })
})
