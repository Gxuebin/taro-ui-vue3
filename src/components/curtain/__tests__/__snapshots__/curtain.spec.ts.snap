// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtCurtain should render default AtCurtain 1`] = `
<view
  class="at-curtain at-curtain--closed"
>
  <view
    class="at-curtain__container"
  >
    <view
      class="at-curtain__body"
    >
      
      
      <view
        class="at-curtain__btn-close at-curtain__btn-close--bottom"
      />
    </view>
  </view>
</view>
`;

exports[`AtCurtain should render prop -- class 1`] = `
<view
  class="test at-curtain at-curtain--closed test"
>
  <view
    class="at-curtain__container"
  >
    <view
      class="at-curtain__body"
    >
      
      
      <view
        class="at-curtain__btn-close at-curtain__btn-close--bottom"
      />
    </view>
  </view>
</view>
`;

exports[`AtCurtain should render prop -- closeBtnPosition 1`] = `
<view
  class="at-curtain at-curtain--closed"
>
  <view
    class="at-curtain__container"
  >
    <view
      class="at-curtain__body"
    >
      
      test
      
      <view
        class="at-curtain__btn-close at-curtain__btn-close--top"
      />
    </view>
  </view>
</view>
`;

exports[`AtCurtain should render prop -- closeBtnPosition 2`] = `
<view
  class="at-curtain at-curtain--closed"
>
  <view
    class="at-curtain__container"
  >
    <view
      class="at-curtain__body"
    >
      
      test
      
      <view
        class="at-curtain__btn-close at-curtain__btn-close--bottom"
      />
    </view>
  </view>
</view>
`;

exports[`AtCurtain should render prop -- isOpened 1`] = `
<view
  class="at-curtain"
>
  <view
    class="at-curtain__container"
  >
    <view
      class="at-curtain__body"
    >
      
      
      <view
        class="at-curtain__btn-close at-curtain__btn-close--bottom"
      />
    </view>
  </view>
</view>
`;

exports[`AtCurtain should render prop -- isOpened 2`] = `
<view
  class="at-curtain at-curtain--closed"
>
  <view
    class="at-curtain__container"
  >
    <view
      class="at-curtain__body"
    >
      
      
      <view
        class="at-curtain__btn-close at-curtain__btn-close--bottom"
      />
    </view>
  </view>
</view>
`;

exports[`AtCurtain should render prop -- style 1`] = `
<view
  class="at-curtain at-curtain--closed"
  style="color: red;"
>
  <view
    class="at-curtain__container"
  >
    <view
      class="at-curtain__body"
    >
      
      
      <view
        class="at-curtain__btn-close at-curtain__btn-close--bottom"
      />
    </view>
  </view>
</view>
`;

exports[`AtCurtain should render prop -- style 2`] = `
<view
  class="at-curtain at-curtain--closed"
  style="color: red;"
>
  <view
    class="at-curtain__container"
  >
    <view
      class="at-curtain__body"
    >
      
      
      <view
        class="at-curtain__btn-close at-curtain__btn-close--bottom"
      />
    </view>
  </view>
</view>
`;
